crates/matrix-sdk/src/sliding_sync/utils.rs

```diff
@@ -0,0 +1,43 @@
+//! Moaaar features for Sliding Sync.
+
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use tokio::task::{JoinError, JoinHandle};
+
+/// Private type to ensure a task is aborted on drop.
+pub(crate) struct AbortOnDrop<T>(JoinHandle<T>);
+
+impl<T> AbortOnDrop<T> {
+    pub fn new(join_handle: JoinHandle<T>) -> Self {
```

Comment:
Let's have just one way of creating `AbortOnDrop`, which is `JoinHandleExt::abort_on_drop`.

```suggestion
    fn new(join_handle: JoinHandle<T>) -> Self {
```

Reply:
:+1: